### PR TITLE
Added new logic to take care of adding products to wishlist

### PR DESCRIPTION
**Pull request by milwoz**

### CODE REVIEW

File: src/Guard/ProductVariantInWishlistGuard.php [hunk 1]
```diff
@@ -17,10 +17,12 @@
 
 final class ProductVariantInWishlistGuard implements ProductVariantInWishlistGuardInterface
 {
-    public function check(WishlistInterface $wishlist, ProductVariantInterface $productVariant): void
+    public function check(WishlistInterface $wishlist, ProductVariantInterface $productVariant): bool
```
Comment (senghe) [inline src/Guard/ProductVariantInWishlistGuard.php:20]: To be honest I don't see any reason of having this class, as we can use the method from WishlistInterface

File: src/Duplicator/WishlistProductsToOtherWishlistDuplicator.php [hunk 2]
```diff
@@ -13,11 +13,15 @@
 
 use BitBag\SyliusWishlistPlugin\Command\Wishlist\WishlistItemInterface;
 use BitBag\SyliusWishlistPlugin\Entity\WishlistInterface;
+use BitBag\SyliusWishlistPlugin\Exception\ProductVariantAlreadyInWishlistException;
 use BitBag\SyliusWishlistPlugin\Facade\WishlistProductFactoryFacadeInterface;
 use BitBag\SyliusWishlistPlugin\Guard\ProductVariantInWishlistGuardInterface;
 use BitBag\SyliusWishlistPlugin\Repository\WishlistRepositoryInterface;
 use Doctrine\Common\Collections\Collection;
+use mysql_xdevapi\Exception;
```
Comment (senghe) [inline src/Duplicator/WishlistProductsToOtherWishlistDuplicator.php:21]: What's this?

**Review by senghe — CHANGES_REQUESTED** [review]

File: src/Duplicator/WishlistProductsToOtherWishlistDuplicator.php (hunk 2 above)
Comment (milwoz) [reply src/Duplicator/WishlistProductsToOtherWishlistDuplicator.php:21]: Added by accident, unused, deleted.

File: src/Guard/ProductVariantInWishlistGuard.php (hunk 1 above)
Comment (milwoz) [reply src/Guard/ProductVariantInWishlistGuard.php]: Done.

**Review by senghe — APPROVED** [review]

**Pull request merged**